metadata/src/utils/validation.rs

```diff
@@ -743,6 +768,36 @@ mod tests {
         assert_eq!(hash, hash_swap);
     }
 
+    #[test]
```

Comment:
It'd also be good now to add a test wherein you basically hash the same type ID twice (or a few times), reusing the same cache for both. The hashes should be identical each time, showing that the cache is truly caching the type hashes and not something else :)